osfoffline/views/start_screen.py

```diff
@@ -21,47 +31,68 @@ def __init__(self):
-                full_name=full_name,
-                osf_id=osf_id,
-                osf_login=user_name,
-                osf_local_folder_path='',
-                oauth_token=oauth_token,
-                osf_password=password
+            resp = yield from osf.make_request(url)
+        except (aiohttp.errors.ClientTimeoutError, aiohttp.errors.ClientConnectionError, concurrent.futures._base.TimeoutError):
+            # No internet connection
+            QMessageBox.warning(
+                None,
+                "Log in Failed",
+                "Unable to connect to server. Check your internet connection or try again later."
+            )
+        except aiohttp.errors.HttpBadRequest:
+            # Invalid credentials
+            QMessageBox.warning(
+                None,
+                "Log in Failed",
+                "Invalid login credentials."
             )
-            user.logged_in = True
-            save(session, user)
+        else:
+            if resp.status == 200:
+                json_resp = yield from resp.json()
+                full_name = json_resp['data']['attributes']['full_name']     # Use one of these later to "Welcome {user}"
+                given_name = json_resp['data']['attributes']['given_name']   # Use one of these later to "Welcome {user}"
+                osf_id = json_resp['data']['id']
+
+                try:
+                    user = session.query(User).filter(User.osf_id == osf_id).one()
+                    user.logged_in = True
+                    save(session, user)
+                except MultipleResultsFound:
+                    logging.warning('multiple users with same username. deleting all users with this username. restarting function.')
+                    for user in session.query(User):
+                        if user.osf_login == osf_id:
+                            session.delete(user)
+                            save(session)
+                    self.log_in()
+                except NoResultFound:
+                    logging.debug('user doesnt exist. Creating user. and logging them in.')
+                    user = User(
+                        full_name=full_name,
+                        osf_id=osf_id,
+                        osf_login='',  # TODO: email goes here when more auth methods are added, not currently returned by APIv2
+                        osf_local_folder_path='',
+                        oauth_token=personal_access_token,
+                    )
+                    user.logged_in = True
+                    save(session, user)
+
+            else:
+                raise OSFAuthError('Invalid auth response: {}'.format(resp.status))
 
             self.close()
 
     def setup_slots(self):
-        self.start_screen.logInButton.clicked.connect(self.log_in)
+        logging.debug('setting up start_screen slots')
+        self.start_screen.logInButton.clicked.connect(lambda: asyncio.get_event_loop().run_until_complete(self.log_in()))
```

Comment:
If I am reading this right, we've written very complicated code that makes an async request (to be non-blocking)... and then converted it into a blocking method?

It seems like there's an extra step there.